File: src/main/java/FlyAway/reservation/dto/ReservationWithoutUserDto.java
package FlyAway.reservation.dto;

import FlyAway.flight.aircraft.CabinClass;
import FlyAway.flight.dto.FlightDto;
import FlyAway.reservation.ReservationStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record ReservationWithoutUserDto(
        LocalDateTime reservationDate,
        BigDecimal price,
        Integer seatNumber,
        CabinClass cabinClass,
        ReservationStatus status,
        FlightDto flightDto
) {
}

File: frontend/src/app/services/models/reservation-without-user-dto.ts
/* tslint:disable */
/* eslint-disable */
import { FlightDto } from '../models/flight-dto';
export interface ReservationWithoutUserDto {
  cabinClass?: 'FIRST' | 'BUSINESS' | 'ECONOMY';
  flightDto?: FlightDto;
  price?: number;
  reservationDate?: string;
  seatNumber?: number;
  status?: 'PENDING' | 'ACTIVE' | 'CANCELLED' | 'EXPIRED' | 'COMPLETED' | 'FAILED';
}
